### PR TITLE
supporting polyline as a pv cut

### LINKED ISSUE #2302: pv image generation along a polyline region (frontend)
This is to extend the pv cut support to include a polyline, hence being able to support more science use cases.

### PR DESCRIPTION
**Pull request by loveluthien**
**Description**

Close #2302. Supporting the polyline region as a PV cut in frontend. This is the companion PR to CARTAvis/carta-backend#1355.

**Checklist**

For linked issues (if there are):
- [x] assignee and label added
- [x] GitHub Project estimate added

For the pull request:
- [x] reviewers and assignee added
- [x] GitHub Project estimate added
- [x] changelog updated / ~no changelog update needed~
- [ ] unit test added (for functions with no dependenies)
- [ ] API documentation added (for public variables and methods in stores)

For dependencies:
- [x] e2e test passing / ~corresponding fix added~ / ~new e2e test created~
- [x] ~protobuf version bumped~ / no protobuf version bumped needed
- [x] ~protobuf updated to the latest dev commit~ / no protobuf update needed
- [ ] `corresponding ICD test fix added (`BackendService` changed)~ / no ICD test fix needed (`BackendService` unchanged)
- [ ] user manual prepared (for large new features)

### CODE REVIEW

**Comment by loveluthien** [comment]
@pford This is the corresponding frontend PR.

**Comment by kswang1029** [comment]
@loveluthien I suggest when a polyline region is selected in the region dropdown, we disable the "Start preview" button. The preview function should only work when a line region is selected. In theory we can also support polyline pv preview but the use case is less clear and it has performance concerns.

**Comment by loveluthien** [comment]
> @loveluthien I suggest when a polyline region is selected in the region dropdown, we disable the "Start preview" button. The preview function should only work when a line region is selected. In theory we can also support polyline pv preview but the use case is less clear and it has performance concerns.

okay, let's do this.

**Review by kswang1029 — APPROVED** [review]
looks good

File: src/components/PvGenerator/PvGeneratorComponent.tsx [hunk 1]
```diff
@@ -284,7 +284,7 @@ export class PvGeneratorComponent extends React.Component<WidgetProps> {
         }
 
         const isAbleToGenerate = this.widgetStore.effectiveRegion && !appStore.animatorStore.animationActive && this.isLineIntersectedWithImage && !this.isLineInOnePixel && this.isValidSpectralRange;
-        const isAbleToGeneratePreview = isAbleToGenerate && this.isCubeSizeBelowLimit;
+        const isAbleToGeneratePreview = isAbleToGenerate && this.isCubeSizeBelowLimit && this.widgetStore.effectiveRegion?.regionType === CARTA.RegionType.LINE;
         const hint = (
```
Comment (YuHsuan-Hwang) [inline src/components/PvGenerator/PvGeneratorComponent.tsx:288]: Perhaps we also adjust the tooltip.

**Review by YuHsuan-Hwang — APPROVED** [review]
Overall looks good. 👍

File: src/components/PvGenerator/PvGeneratorComponent.tsx (hunk 1 above)
Comment (loveluthien) [reply src/components/PvGenerator/PvGeneratorComponent.tsx:288]: done

**Pull request merged**